### PR TITLE
Modflow6

### PR DESCRIPTION
**Pull request by scopatz**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

CC @dharhas

### CODE REVIEW

**Comment by scopatz** [comment]
@conda-forge/staged-recipes This is a unix-only recipe that is ready for review!

File: recipes/modflow6/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    # use inherited environment variables
+    - makefile.patch
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('fortran') }}
```
Comment (marcelotrevisani) [inline recipes/modflow6/meta.yaml]: don't you need the ``C`` compiler as well?
```suggestion
    - {{ compiler('fortran') }}
    - {{ compiler('c') }}
```

File: recipes/modflow6/makefile.patch [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+-FC = gfortran
+-FFLAGS = -O2 -fbacktrace 
++#FC = gfortran
++#FFLAGS = -O2 -fbacktrace 
+ 
+ # Define the C compile flags
+-CC = gcc
+-CFLAGS = -O2 -D_UF 
++#CC = gcc
++#CFLAGS = -O2 -D_UF
```
Comment (isuruf) [inline recipes/modflow6/makefile.patch]: Is `-D_UF` not needed?

File: recipes/modflow6/makefile.patch [hunk 3]
```diff
@@ -0,0 +1,23 @@
+--- make/makefile.orig	2020-02-13 13:12:32.254482296 -0500
++++ make/makefile	2020-02-13 13:14:10.197751134 -0500
+@@ -42,15 +42,15 @@
+ .SUFFIXES: .c .cpp .f .f90 .F90 .fpp .o
+ 
+ # Define the Fortran compile flags
+-FC = gfortran
+-FFLAGS = -O2 -fbacktrace 
++#FC = gfortran
++#FFLAGS = -O2 -fbacktrace
```
Comment (isuruf) [inline recipes/modflow6/makefile.patch]: Same here with `-fbacktrace`.

File: recipes/modflow6/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  commands:
+    - mf6 --help
+
+about:
+  home: https://www.usgs.gov/software/modflow-6-usgs-modular-hydrologic-model
+  license: CC0
+  license_family: CC
+  license_file: LICENSE.md
+  summary: 'Open-source groundwater flow model'
+  doc_url: https://water.usgs.gov/water-resources/software/MODFLOW-6/mf6io_6.1.0.pdf
```
Comment (marcelotrevisani) [inline recipes/modflow6/meta.yaml]: Maybe?
```suggestion
  doc_url: https://water.usgs.gov/water-resources/software/MODFLOW-{{ version[0] }}/mf6io_{{ version}}.pdf
```

File: recipes/modflow6/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
+
+test:
+  commands:
+    - mf6 --help
+
+about:
+  home: https://www.usgs.gov/software/modflow-6-usgs-modular-hydrologic-model
+  license: CC0
```
Comment (isuruf) [inline recipes/modflow6/meta.yaml]: Can you use the SPDX identifier?

Comment (scopatz) [reply recipes/modflow6/meta.yaml]: done

**Comment by scopatz** [comment]
This is ready for another review!

**Comment by scopatz** [comment]
Tests pass! thanks @isuruf @marcelotrevisani!

**Pull request merged**